### ISSUE TITLE
pytest_vulture.setup_manager.EntryPointFileError: an entry point file is missing in the setup.py/pyproject.toml file

### ISSUE THREAD

**Issue by nth10sd** [Gatewatcher/pytest-vulture#19]
```
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_pylint/plugin.py:223: PytestRemovedIn9Warning: The (path: py.path.local) argument is deprecated, please use (file_path: pathlib.Path)
see https://docs.pytest.org/en/latest/deprecations.html#py-path-local-arguments-for-hooks-replaced-with-pathlib-path
  def pytest_collect_file(self, path, parent):
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/main.py", line 281, in wrap_session
INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_hooks.py", line 513, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_callers.py", line 122, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>     ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/logging.py", line 782, in pytest_sessionstart
INTERNALERROR>     return (yield)
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_callers.py", line 103, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/plugin.py", line 44, in pytest_sessionstart
INTERNALERROR>     session.vulture = VultureManager(root_dir, reader)
INTERNALERROR>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/vulture/manager.py", line 32, in __init__
INTERNALERROR>     self.__setup_manager = SetupManager(self._config)
INTERNALERROR>                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/setup_manager.py", line 48, in __init__
INTERNALERROR>     self._generate_entry_points(content)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/setup_manager.py", line 100, in _generate_entry_points
INTERNALERROR>     self._find_py_project_toml(content)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/setup_manager.py", line 94, in _find_py_project_toml
INTERNALERROR>     self.__parse_entry_point_line(entry_point.replace(" ", "").replace('"', ""), root_paths)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/setup_manager.py", line 126, in __parse_entry_point_line
INTERNALERROR>     self.__check_entry_points()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pytest_vulture/setup_manager.py", line 151, in __check_entry_points
INTERNALERROR>     raise EntryPointFileError(message=f"{path.as_posix()} can not be found. {_ENTRY_POINT_MSG}")
INTERNALERROR> pytest_vulture.setup_manager.EntryPointFileError: an entry point file is missing in the setup.py/pyproject.toml file : /home/runner/work/vanctransit/vanctransit/vanctransit/_vanctransit.py can not be found. 
INTERNALERROR> You can add the source_path parameter in the settings or disable with check-entry-points=0
```

Some of my projects use [maturin](https://github.com/PyO3/maturin) and they hit this issue.

E.g. the python files are under an additional `python/` subfolder (which is not the package name) on the root directory:

```
python/
  -> rest of the python source code
```

`pytest-vulture` 2.2.1 seems to work correctly when the python files are under a subfolder which is named after the package.

Regression from #15, @MoranAbadieGatewatcher?

**Comment by MoranAbadieGatewatcher** [comment]
can you try to add source-path = "python" ? in [tool.vulture] ?

**Comment by nth10sd** [comment]
```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/_pytest/main.py", line 281, in wrap_session
INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pluggy/_hooks.py", line 513, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 122, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>     ~~~~~~~~~~~~~~^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/_pytest/logging.py", line 782, in pytest_sessionstart
INTERNALERROR>     return (yield)
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pluggy/_callers.py", line 103, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/plugin.py", line 44, in pytest_sessionstart
INTERNALERROR>     session.vulture = VultureManager(root_dir, reader)
INTERNALERROR>                       ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/vulture/manager.py", line 32, in __init__
INTERNALERROR>     self.__setup_manager = SetupManager(self._config)
INTERNALERROR>                            ~~~~~~~~~~~~^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/setup_manager.py", line 48, in __init__
INTERNALERROR>     self._generate_entry_points(content)
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/setup_manager.py", line 100, in _generate_entry_points
INTERNALERROR>     self._find_py_project_toml(content)
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/setup_manager.py", line 94, in _find_py_project_toml
INTERNALERROR>     self.__parse_entry_point_line(entry_point.replace(" ", "").replace('"', ""), root_paths)
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/setup_manager.py", line 126, in __parse_entry_point_line
INTERNALERROR>     self.__check_entry_points()
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~^^
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.13.0/x64/lib/python3.13/site-packages/pytest_vulture/setup_manager.py", line 151, in __check_entry_points
INTERNALERROR>     raise EntryPointFileError(message=f"{path.as_posix()} can not be found. {_ENTRY_POINT_MSG}")
INTERNALERROR> pytest_vulture.setup_manager.EntryPointFileError: an entry point file is missing in the setup.py/pyproject.toml file : /home/runner/work/vanctransit/vanctransit/python/vanctransit/_vanctransit.py can not be found. 
INTERNALERROR> You can add the source_path parameter in the settings or disable with check-entry-points=0
```

Tried it, same issue?

**Comment by MoranAbadieGatewatcher** [comment]
Sorry about that. Could you please provide me with the entry point in the pyproject.toml file that's causing the issue? If you prefer, you can rename it for privacy. Ideally, this entry point should point to vanctransit/_vanctransit.py. Can you also give me the right path of vanctransit/_vanctransit.py in your repo please ?

The code's intent (though not implemented very cleanly, I admit) is to look at [project.scripts] and retrieve the script names to exclude them. This ensures Vulture recognizes them as entry points and avoids flagging them as unused.

In the meantime, you can set check-entry-points=0 under [tool.vulture] to temporarily bypass the error.

**Comment by nth10sd** [comment]
Please feel free to take a look [at the repo](https://github.com/nth10sd/vanctransit).

**Comment by MoranAbadieGatewatcher** [comment]
Thank you,

maybe there is some python magic that I do not get but : https://github.com/nth10sd/vanctransit/blob/main/pyproject.toml#L54 does not seems to point to anything right ?

I do not see any _vanctransit in https://github.com/nth10sd/vanctransit/tree/main/python/vanctransit :)

**Comment by MoranAbadieGatewatcher** [comment]
oh it is in https://github.com/nth10sd/vanctransit/blob/main/src/lib.rs#L17, so yes pytest-vulture can not check that, so the solution might be to add check-entry-points=0

**Comment by nth10sd** [comment]
I tried to add `check-entry-points=0` to `[tool.vulture]` and the error is [still there in this run](https://github.com/nth10sd/vanctransit/actions/runs/12007355852).

**Comment by nth10sd** [comment]
I think I'm not setting the setting correctly? How should I set it?

**Comment by MoranAbadieGatewatcher** [comment]
ok its a bug I will fix it asap, thank you

**Comment by MoranAbadieGatewatcher** [comment]
its should be fixed on 2.2.2

**Comment by nth10sd** [comment]
Thanks! I verified that this issue is now fixed with 2.2.2.
